core/tracker/tracker.go

```diff
@@ -132,17 +132,61 @@ func analyseDutyFailed(duty core.Duty, es []event) (bool, component, string) {
 	})
 
 	if len(events) == 0 {
-		return false, sentinel, "No events to analyse"
+		return false, sentinel
 	}
 
-	if events[0].component == sigAgg {
-		// Duty completed successfully
+	c := events[0].component
+	if c == sigAgg {
+		return false, sigAgg
+	}
+
+	return true, c + 1
+}
+
+// analyseDutyFailed detects if the given duty failed. It returns false if the duty didn't fail, i.e., the duty didn't get stuck and completes the sigAgg component.
+// It returns true if the duty failed. It also returns the component where the duty got stuck and a human friendly error message explaining why.
+func analyseDutyFailed(duty core.Duty, allEvents map[core.Duty][]event) (bool, component, string) {
+	var (
+		failed bool
+		comp   component
+		msg    string
+	)
+
+	failed, comp = dutyFailedComponent(allEvents[duty])
+	if !failed {
 		return false, sigAgg, ""
 	}
 
-	// TODO(xenowits): Improve message to have more specific details.
-	//  Ex: If the duty got stuck during validatorAPI, we can say "the VC didn't successfully submit a signed duty").
-	return true, events[0].component + 1, fmt.Sprintf("%s failed in %s component", duty.String(), (events[0].component + 1).String())
+	switch comp {
+	case fetcher:
+		msg = "couldn't fetch duty data from the beacon node"
+
+		if duty.Type == core.DutyProposer || duty.Type == core.DutyBuilderProposer {
+			// Proposer duties may fail if core.DutyRandao fails
+			randaoFailed, randaoComp := dutyFailedComponent(allEvents[core.NewRandaoDuty(duty.Slot)])
+			if randaoFailed {
+				if randaoComp == parSigDBThreshold {
+					msg = "couldn't propose block due to insufficient partial randao signatures"
+				} else {
+					msg = "couldn't propose block since randao duty failed"
+				}
+			}
+		}
+	case consensus:
+		msg = "consensus algorithm didn't complete"
```

Comment:
maybe make these constants, so comparison and tests and updates are trivial.

Reply:
agreed, great idea